mkosi/__init__.py

```diff
@@ -4279,6 +4279,9 @@ def install_unified_kernel(
             if root_hash:
                 option = "usrhash" if args.usr_only else "roothash"
                 boot_options = f"{boot_options} {option}={root_hash}"
+            elif args.usr_only and not args.verity:
+                # Leave systemd-gpt-auto-generator to mount the root partition
+                pass
```

Comment:
How does this accomplish what you want? `root=` is only set if the image has an image ID. If you don't have an image ID, `root=` won't be set

Reply:
Sorry, I do have an image ID set, but you want to use it as `mount.usr=/dev/disk/by-partlabel/<ImageID>_<ImageVersion>` (as is set currently) without also setting `root=PARTLABEL=<ImageID>_<ImageVersion>` for this particular case.

Reply:
What if we just set `mount.usr` here instead of `root` if `usr_only == True`? Does that also address the issue?

Reply:
`mount.usr` is already set over [here](https://github.com/systemd/mkosi/blob/f9ebc563877583d9d49d8d566ea0a65c33f01b7d/mkosi/__init__.py#L6864) so by the time it comes round to this bit of code it's only `root=` that is being set or skipped. The grand total is a kernel command line with `mount.usr` and no `root` parameter.

Reply:
Yes, you'll end up mount.usr set 2x but the last occurrence will override so if we set `mount.usr` here instead of `root`, things should still work.

The entire calculation of kernel command line arguments needs to be reworked but the most "correct" fix atm that doesn't touch too much code is to set `mount.usr` here if `args.usr_only == True`

Reply:
OK, I've added `mount.usr` instead of skipping.